Undo removal of getAllObservationKeys in SummaryObservationCollector

**Describe the bug**
Semeio uses this function in a test so undo the removal

```
=================================== FAILURES ===================================
__________________________ test_ahmanalysis_run_field __________________________

test_data_root = '/tmp/jenkins-komodo-f_scout_ci/workspace/unit-tests/semeio/test-root/tests/legacy_test_data'
grid_prop = <function grid_prop.<locals>.wrapper at 0x7f6248be8620>

    @pytest.mark.usefixtures("setup_tmpdir")
    def test_ahmanalysis_run_field(test_data_root, grid_prop):
        """test data_set with scalar and Field parameters"""
        test_data_dir = os.path.join(test_data_root, "snake_oil")
    
        shutil.copytree(test_data_dir, "test_data")
        os.chdir(os.path.join("test_data"))
        os.makedirs("fields")
        grid = EclGridGenerator.createRectangular((10, 12, 5), (1, 1, 1))
        for iens in range(10):
            grid_prop("PERMX", 10, grid.getGlobalSize(), f"fields/permx{iens}.grdecl")
            grid_prop("PORO", 0.2, grid.getGlobalSize(), f"fields/poro{iens}.grdecl")
    
        res_config = ResConfig("snake_oil_field.ert")
        res_config.convertToCReference(None)
        ert = EnKFMain(res_config)
    
        ahmanalysis.AhmAnalysisJob(ert).run(prior_name="default")
    
        # assert that this returns/generates the delta field parameter
        gen_obs_list = GenDataObservationCollector.getAllObservationKeys(ert)
>       summary_obs_list = SummaryObservationCollector.getAllObservationKeys(ert)
E       AttributeError: type object 'SummaryObservationCollector' has no attribute 'getAllObservationKeys'

/tmp/jenkins-komodo-f_scout_ci/workspace/unit-tests/semeio/test-root/tests/jobs/ahm_analysis/test_integration.py:91: AttributeError
```

--- a/res/enkf/export/gen_kw_collector.py
+++ b/res/enkf/export/gen_kw_collector.py
@@ -18,6 +18,11 @@ class GenKwCollector:
         return bool_vec.createActiveList()
 
     @staticmethod
+    def getAllGenKwKeys(ert):
+        """@rtype: list of str"""
+        return ert.getKeyManager().genKwKeys()
+
+    @staticmethod
     def loadAllGenKwData(ert: EnKFMain, case_name, keys=None, realization_index=None):
         """
         @type ert: EnKFMain

--- a/res/enkf/export/misfit_collector.py
+++ b/res/enkf/export/misfit_collector.py
@@ -13,6 +13,11 @@ class MisfitCollector:
         return [index for index, element in enumerate(ens_mask) if element]
 
     @staticmethod
+    def getAllMisfitKeys(ert, sort_keys=True):
+        """@rtype: list of str"""
+        return ert.getKeyManager().misfitKeys(sort_keys=sort_keys)
+
+    @staticmethod
     def loadAllMisfitData(ert: EnKFMain, case_name) -> DataFrame:
         """
         @type ert: EnKFMain

--- a/res/enkf/export/summary_collector.py
+++ b/res/enkf/export/summary_collector.py
@@ -6,6 +6,11 @@ from res.enkf.enums import RealizationStateEnum
 
 class SummaryCollector:
     @staticmethod
+    def getAllSummaryKeys(ert):
+        """@rtype: list of str"""
+        return ert.getKeyManager().summaryKeys()
+
+    @staticmethod
     def createActiveList(ert, fs):
         state_map = fs.getStateMap()
         ens_mask = state_map.selectMatching(RealizationStateEnum.STATE_HAS_DATA)

--- a/res/enkf/export/summary_observation_collector.py
+++ b/res/enkf/export/summary_observation_collector.py
@@ -4,6 +4,14 @@ from res.enkf import EnKFMain
 
 class SummaryObservationCollector:
     @staticmethod
+    def getAllObservationKeys(ert: EnKFMain):
+        """
+        @type ert: EnKFMain
+        @rtype: list of str
+        """
+        return ert.getKeyManager().summaryKeysWithObservations()
+
+    @staticmethod
     def loadObservationData(ert: EnKFMain, case_name, keys=None):
         """
         @type ert: EnKFMain

--- a/tests/libres_tests/res/enkf/export/test_summary_observation_collector.py
+++ b/tests/libres_tests/res/enkf/export/test_summary_observation_collector.py
@@ -31,6 +31,11 @@ class SummaryObservationCollectorTest(ResTest):
                 SummaryObservationCollector.summaryKeyHasObservations(ert, "FOPT")
             )
 
+            keys = SummaryObservationCollector.getAllObservationKeys(ert)
+            self.assertTrue("FOPR" in keys)
+            self.assertTrue("WOPR:OP1" in keys)
+            self.assertFalse("WOPR:OP2" in keys)
+
             data = SummaryObservationCollector.loadObservationData(ert, "default_0")
 
             self.assertFloatEqual(data["FOPR"]["2010-01-10"], 0.001696887)

